cmd/juju/model/switchgeneration.go

```diff
@@ -104,12 +105,23 @@ func (c *switchGenerationCommand) Run(ctx *cmd.Context) error {
 	}
 	defer client.Close()
 
-	// TODO (hml) 20-12-2018
-	// update to check err when SwitchGeneration() is implemented in the
-	// apiserver.
-	client.SwitchGeneration(c.generation)
+	_, modelDetails, err := c.ModelCommandBase.ModelDetails()
+	if err != nil {
+		return errors.Annotate(err, "getting model details")
+	}
+
+	modelTag := names.NewModelTag(modelDetails.ModelUUID)
+	if err = client.SwitchGeneration(modelTag, c.generation); err != nil {
+		return err
+	}
 
-	msg := fmt.Sprintf("changes dropped and target generation set to %s\n", c.generation)
+	var msg string
+	switch c.generation {
+	case "next":
+		msg = fmt.Sprintf("target generation set to %s\n", c.generation)
+	case "current":
+		msg = fmt.Sprintf("changes dropped and target generation set to %s\n", c.generation)
```

Comment:
Is "dropped" the right verbiage to use? Might give some operators a moment of panic.

Reply:
Must be copy paste error from cancel generation.